Support sink to URI for sources creation

invincibleJai opened the pull request:
**Fixes**: 
https://issues.redhat.com/browse/ODC-4241

**Analysis / Root cause**: 
User is unable to sink a source to URI along with ksvc/Channel/Broker

**Solution Description**: 
- Show URI as sink option under radio group along with Resources
- Resources will be pre-selected
- Should be able to create Source with Sink to URI
- Added yup validations
- Remove error state shown in case of no resources (KSVC/Channel/Broker) present as no more needed)

**Screen shots / Gifs for design review**: 
<!-- If change affects UI in any way, tag @openshift/team-devconsole-ux and add screenshots/gifs  -->
- Resource preselected
![image](https://user-images.githubusercontent.com/5129024/87169350-77baf400-c2ed-11ea-829f-be7a8cdcf57a.png)


- Sink URI
![image](https://user-images.githubusercontent.com/5129024/87169425-9a4d0d00-c2ed-11ea-9788-02f9bd72dda8.png)



- Error State (Invalid URI)
![image](https://user-images.githubusercontent.com/5129024/87169528-c072ad00-c2ed-11ea-9502-e69a343f052c.png)


- Inline Alert if no resources(KSVC, Channel, Broker) exists 
![image](https://user-images.githubusercontent.com/5129024/87171576-7b03af00-c2f0-11ea-8e9b-327a5cd1658d.png)




cc @openshift/team-devconsole-ux @rachael-phillips 

**Unit test coverage report**: 

![image](https://user-images.githubusercontent.com/5129024/86444560-883bff00-bd2e-11ea-9e86-847800da09d2.png)



**Browser conformance**: 
<!-- To mark tested browsers, use [x] -->
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

karthikjeeyar commented on frontend/packages/console-shared/src/components/formik-fields/RadioGroupField.scss:
```diff
@@ -19,3 +19,9 @@
 
   }
 }
+
+.pf-c-radio {
+  &__description {
+    width: 100%;
+  }
+}
```
Move this styles definition into the scope of `ocs-radio-group-field` block, so it wont override the default pf-radio-button styles.
```suggestion

.pf-c-radio {
  &__description {
    width: 100%;
  }
 }
}
```

karthikjeeyar commented on frontend/packages/knative-plugin/src/components/add/eventSource-validation-utils.ts:
```diff
@@ -7,9 +8,26 @@ import {
+const sinkServiceSchema = yup
+  .object()
+  .when('sinkType', {
+    is: 'resource',
```
```suggestion
    is: SinkType.Resource,
```

karthikjeeyar commented on frontend/packages/knative-plugin/src/components/add/eventSource-validation-utils.ts:
```diff
@@ -7,9 +8,26 @@ import {
+    }),
+  })
+  .when('sinkType', {
+    is: 'uri',
```
```suggestion
    is: SinkType.Uri,
```

rachael-phillips commented:
The basic design is accurate in this PR. We will need to go back and open a PR for the final copy.

invincibleJai commented:
@rachael-phillips have updated PR based on final design
- updated help text
- added Inline Alert if no resources(KSVC, Channel, Broker) exists and disable resource dropdown

PTAL, when you get sometime

invincibleJai commented:
/test images

rachael-phillips commented:
This is look great! Copy under Sink should be "Add a Sink to route this Event Source to a Channel, Broker, Knative Service or another route." 

Capitalization has been updated under the Resource field. "This resource will be the Sink for the Event Source." 

The "a" was missing in alert. Copy in alert should read "Select the URI option, or exit this form and create a Knative Service, Broker, or Channel first."

invincibleJai commented:
@rachael-phillips have updated the texts and screenshot's as well in description. Thanks , PTAL :)

rachael-phillips commented:
LGTM!

karthikjeeyar commented:
verified locally
/lgtm

karthikjeeyar commented:
/lgtm

invincibleJai commented:
/assign @christianvogt

abhi-kn commented:
/retest

christianvogt commented:
@beaumorley 
Should we consider defaulting the form to URI if there are no services to sink to?
![image](https://user-images.githubusercontent.com/14068621/87827390-d3830100-c848-11ea-8abe-058d11667dd8.png)

christianvogt commented:
/lgtm
/approve

rachael-phillips commented:
@christianvogt yes, the URI field should be the default if there are no resources. This was specified in the [design doc](https://docs.google.com/document/d/102SirBYyfQQRl1hxNTNsp3xAMaS3AN-3bUljYaN7uOU/edit?usp=sharing) which reads "If the user manually selects a resource the field should appear as disabled. An inline alert should appear with a message that describes for the user that they should try the URI field instead."
